content/blog/cloud-native-secret-management-with-pulumi-esc-and-external-secrets-operator/index.md

```diff
@@ -0,0 +1,419 @@
+                    secretKeyRef: {
+                        name: "esc-secret-store",
+                        key: "hello",
+                    }
+                }
+            }
+        ]
+    },
+});
+
+const externalSecretPodInfo = new k8s.apiextensions.CustomResource("external-secret-podinfo", {
+    apiVersion: "external-secrets.io/v1beta1",
+    kind: "ExternalSecret",
+    metadata: {
+        name: "esc-secret-store",
+        namespace: podInfo.namespace,
+    },
+    spec: {
+        dataFrom: [
+            {
+                extract: {
+                    conversionStrategy: "Default",
+                    key: "app",
+                }
+            }
+        ],
+        refreshInterval: "10s",
+        secretStoreRef: {
+            kind: clusterSecretStore.kind,
+            name: clusterSecretStore.metadata.name,
+        }
+    },
+});
+```
+
+This will deploy the `PodInfo` application and create an `ExternalSecret` resource that references the secret from the Pulumi ESC project. We mount the secret into the pod as an environment variable by using the `extraEnvs` field in the Helm chart.
+
+{{% notes type="info" %}}
+
+In a real-world application, you might use alternate ways to reference the secret, like mounting it as a volume or using `fromEnv` field in the Helm chart. It depends on your application and how you want to consume the secret.
+
+{{% /notes %}}
+
+After deploying the stack, you should see that the `PodInfo` application is running. To access the application, we can `port-forward` the service by running:
+
+```bash
+kubectl port-forward svc/podinfo 9898:9898 -n podinfo
+```
+
+Our `PodInfo` demo application provides a large selection of different [endpoints](https://github.com/stefanprodan/podinfo) to interact with. We are interested in the `/env` endpoint to check if the secret was successfully mounted into the pod. Let's try that:
+
+```bash
+curl -s  localhost:9898/env | grep FROM
+  "FROM_ESC_VIA_ESO=world",
+  "FROM_ESC_VIA_ESO_2=world",
+```
+
+And here we see secret values from the ESC project, proxied by ESO, and passed to the PodInfo application as environment variables. That means the secret was successfully mounted into the pod and the application is running as expected.
+
+### Step 4: Clean Up
+
+After you are done with the demo, you can clean up the resources by running:
+
+```bash
+pulumi destroy
+```
+
+## Conclusion
+
+The combination of Pulumi ESC and External Secrets Operator provides a secure and efficient way to manage secrets in a cloud-native environment, improving security, efficiency, and compliance when consuming secrets in your Kubernetes cluster. By using Pulumi ESC, you can manage secrets and configurations in a centralized way, and with ESO, you now have a streamlined process to synchronize secrets into your Kubernetes cluster.
+
+Start using Pulumi ESC and ESO today by creating an account on the Pulumi Cloud Console and begin managing your secrets in a secure and efficient way.
+
```

Comment:
Add something like "Don't have a Pulumi account yet. Click here to create an account and starting using Pulumi ESC"